pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -169,6 +171,32 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 		return nil, err
 	}
 
+	// Replace the standard k8s emptyDir volume plugin with a wrapper version
+	// which offers XFS quota functionality, but only if the node config
+	// specifies an empty dir quota to apply to projects:
+	if options.ProjectEmptyDirQuota != nil {
+		glog.V(2).Info("Replacing empty-dir volume plugin with quota wrapper")
+		wrappedEmptyDirPlugin := false
+		for idx, plugin := range cfg.VolumePlugins {
+			// Can't really do type checking or use a constant here as they are not exported:
+			if plugin.Name() == "kubernetes.io/empty-dir" {
```

Comment:
@deads2k @liggitt you guys ok with the composition rather than inheritance compromise above? Should eliminate the silent breakage during a rebase concern. Going to start work on it momentarily if so.

Re: CanSupport, sounds good I will try that out.

This plugin can't do much without an instance of an emptyDir plugin that can actually do the work, so we'd need to instantiate one ourselves if it's not already in the list. Afaik you can't configure the list of plugins to be enabled today, so if emptyDir were to not be there we probably want to be looking into why, it might be another situation where we should fail fast. I will take a look though and see.